examples/decoding/plot_ems_filtering.py

```diff
@@ -57,38 +60,70 @@
 
 epochs = mne.Epochs(raw, events, event_ids, tmin, tmax, picks=picks,
-                    baseline=None, reject=reject)
-
-# Let's equalize the trial counts in each condition
-epochs.equalize_event_counts(epochs.event_id, copy=False)
-
-# compute surrogate time series
-surrogates, filters, conditions = compute_ems(epochs, ['AudL', 'VisL'])
-
-times = epochs.times * 1e3
+                    baseline=None, reject=reject, preload=True)
+epochs.drop_bad()
+epochs.pick_types(meg='grad')
+
+# Setup the data to use it a scikit-learn way:
+X = epochs.get_data()  # The MEG data
+y = epochs.events[:, 2]  # The conditions indices
+n_epochs, n_channels, n_times = X.shape
+
+# Initialize EMS transformer
+ems = EMS()
+
+# initialize the variables of interest
+X_transform = np.zeros((n_epochs, n_times))  # Data after EMS transformation
+filters = list()  # Spatial filters at each time point
+
+# In the original paper, the cross-validation is a leave-one-out. However,
+# we recommend using a Stratified KFold, as LOO is known to generate biases.
+for train, test in StratifiedKFold(y):
+    # In the original paper, the z-scoring is applied outside the CV.
+    # However, we recommend to apply this preprocessing inside the CV.
+    # Note that such scaling should be done separately for each channels if the
+    # data contains multiple channel types.
+    # Note that X_scaled does not have any unit anymore.
+    X_scaled = X / np.std(X[train])
+
+    # Fit and store the spatial filters
+    ems.fit(X_scaled[train], y[train])
+    filters.append(ems.filters_)
+
+    # Generate the transformed data
+    X_transform[test] = ems.transform(X_scaled[test])
+
+# Average the spatial filters
+filters = np.mean(filters, axis=0)
+
+# Plot individual trials
 plt.figure()
 plt.title('single trial surrogates')
-plt.imshow(surrogates[conditions.argsort()], origin='lower', aspect='auto',
-           extent=[times[0], times[-1], 1, len(surrogates)],
+plt.imshow(X_transform[y.argsort()], origin='lower', aspect='auto',
+           extent=[epochs.times[0], epochs.times[-1], 1, len(X_transform)],
            cmap='RdBu_r')
 plt.xlabel('Time (ms)')
 plt.ylabel('Trials (reordered by condition)')
 
+# Plot average response
 plt.figure()
 plt.title('Average EMS signal')
-
-mappings = [(k, v) for k, v in event_ids.items() if v in conditions]
+mappings = [(key, value) for key, value in event_ids.items()]
 for key, value in mappings:
-    ems_ave = surrogates[conditions == value]
-    ems_ave *= 1e13
-    plt.plot(times, ems_ave.mean(0), label=key)
+    ems_ave = X_transform[y == value]
+    plt.plot(epochs.times, ems_ave.mean(0), label=key)
 plt.xlabel('Time (ms)')
-plt.ylabel('fT/cm')
+plt.ylabel('a.u.')
 plt.legend(loc='best')
-
-
-# visualize spatial filters across time
 plt.show()
-evoked = epochs.average()
-evoked.data = filters
+
+# Visualize spatial filters across time
+evoked = EvokedArray(filters, epochs.info, tmin=epochs.tmin)
 evoked.plot_topomap(ch_type=ch_type)
+
```

Comment:
insert here line like this:

#############################################################

so it renders on the doc and notebook output as a separate section and images are then plotted at the end of each section rather than at the top.